cogs/accounts.py

```diff
@@ -222,100 +224,104 @@ async def stats(self,
+        if user_profile:
+            coins_string = ', '.join(
+                [x["assetCode"].upper() for x in self.bot.backoffice.token_manager.get_registered_tokens()])
+
+            description = ':warning: To deposit funds to your Discord wallet, you will need to withdraw from your ' \
+                          'preferred wallet(GUI, CLI) to the address and deposit ID provided below. ' \
+                          'If done incorrectly this can result in funds being lost and irrecoverable! ' \
+                          'Crypto Link staff are not responsible for any mistakes made. :warning:'
+
+            deposit_embed = Embed(title='How to deposit',
+                                  colour=Colour.dark_orange(),
+                                  description=description)
+            deposit_embed.add_field(name=':gem: Supported Cryptocurrencies/Tokens for deposit:gem: ',
+                                    value=f'```{coins_string}```',
+                                    inline=False)
+            deposit_embed.add_field(
+                name=f'Deposit Details',
+                value=f'\n:map: Public Address :map: \n'
+                      f'```{self.backoffice.stellar_wallet.public_key}```\n'
+                      f':compass: MEMO :compass:\n'
+                      f'```{user_profile["stellarDepositId"]}```',
+                inline=False)
+            deposit_embed.add_field(name=':warning: **__Warning__** :warning:',
+                                    value='Make sure to include and provide an appropriate  **__MEMO__** as text '
+                                          'and Wallet address for each currency , otherwise your deposit will be lost!',
+                                    inline=False)
+            deposit_embed.add_field(name=":printer: QR Code :printer: ",
+                                    value=f'Below is your personal QR code including your deposit address and '
+                                          f'MEMO. Scan it with a mobile application supporting QR codes. Be sure to '
+                                          f'double check the data once you scan it.',
+                                    inline=False)
+
+            memo = user_profile["stellarDepositId"]
+            uri = self.backoffice.stellar_wallet.generate_uri(address=self.backoffice.stellar_wallet.public_key,
+                                                              memo=memo)
+            image = pyqrcode.create(content=uri, error='L')
+            image.png(file=f'{interaction.message.author.id}.png', scale=6, module_color=[0, 255, 255, 128],
+                      background=[17, 17, 17],
+                      quiet_zone=4)
+            qr_to_send = File(f'{interaction.message.author.id}.png')
+
+            deposit_embed.set_image(url=f"attachment://{interaction.message.author.id}.png")
+            deposit_embed.set_footer(text=f'/wallet deposit qr -> Only QR')
```

Comment:
You could branch out this part into separated function inside the class and return the discord embed from function which will be than used to send the details, or you could actually branch out the whole functionality in custom function called for example:
async def send_qr_code_picture(self, REGUIRED PARAMETERS)
....

Try it out and if you cant figure it out ill check in the morning